src/main/java/spoon/support/reflect/code/CtLambdaImpl.java

```diff
@@ -124,6 +119,9 @@ public String getSimpleName() {
 			lambdaExecutableMethod = lambdaTypeMethods.iterator().next();
 		} else {
 			for (CtMethod<?> method : lambdaTypeMethods) {
+				if (getFactory().Method().OBJECT_METHODS.contains(method)) {
```

Comment:
Thanks for the incredibly fast feedback!

I just tried using `CtMethod.isOverriding`.

```java
for (CtMethod<?> method : lambdaTypeMethods) {
    if (getFactory().Method().OBJECT_METHODS.stream().anyMatch(method::isOverriding)) {
        continue;
    }
...
```

Unfortunately the method does not return `true` since the Interface is no subtype of `java.lang.Object` ([Reference](https://github.com/INRIA/spoon/blob/master/src/main/java/spoon/support/visitor/ClassTypingContext.java#L274-L277)).

It seems the only solution is comparing method signatures, as you said.